### DISCUSSION TITLE
ZeroMQ implementation in Fprime

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#2072] [Q&A]
Hello!

I have been teaching myself about ZeroMQ and wanted to know if anyone had made a ZeroMQ driver in Fprime? From my minimal knowledge, it seems like some potential use cases could be using ZMQ for communication between nodes in a network of multiple Fprime deployments or even as simple as connecting to the GDS via ZMQ.

Does anyone have any thoughts on how they would implement a ZeroMQ driver in Fprime or thoughts on any other cool potential use cases?

**Comment by timcanham** [comment]
We have an old implementation here that we might be able to put in fprime-community. It implements a one-way [hub](https://nasa.github.io/fprime/v3.1.0/UsersGuide/best/hub-pattern.html) pattern where the ZMQ sender has [serialized](https://nasa.github.io/fprime/v3.1.0/UsersGuide/user/port-comp-top.html) imputs and the ZMQ receiver has serialized output ports. It uses the old XML interface specifications and an old version of ZMQ, but it could probably be updated if you wanted to try it.

**Reply by ebenezerarunkumar** [reply]
that would be great! would love to see how y'all implemented it! let me know if I can be of any help for updating the existing code to the new version of fprime
